Modules/_sqlite/connection.c

```diff
@@ -432,48 +436,75 @@ free_callback_contexts(pysqlite_Connection *self)
 static void
 remove_callbacks(sqlite3 *db)
 {
+    if (db == NULL) {
+        return;
+    }
     sqlite3_trace_v2(db, SQLITE_TRACE_STMT, 0, 0);
     sqlite3_progress_handler(db, 0, 0, (void *)0);
     (void)sqlite3_set_authorizer(db, NULL, NULL);
```

Comment:
Can any of these calls fail? In what state it leaves the db?

Reply:
Since we never call these APIs with an invalid database pointer, these APIs cannot fail. `sqlite3_trace_v2` and `sqlite3_set_authorizer` always return `SQLITE_OK`, so I guess we could add an `assert` for those, just to be sure. The progress handler API returns void, so there's nothing to do there. We can also assert we are giving a valid database pointer, in case the runtime SQLite library was compiled with `SQLITE_ENABLE_API_ARMOR`.

Reply:
I added asserts. Perhaps a clarifying comment is needed.

Reply:
I added a short comment.